pipeline/load/load.py

```diff
@@ -30,22 +30,113 @@ def get_db_connection(config) -> connection:
 
 def format_release_date_dt(game_data: list[list]) -> list[list]:
     """Given our game data, we format the release date as a datetime object."""
+
     for game in game_data:
         release_date = game[5]
         game[5] = datetime.strptime(release_date, "%Y-%m-%d %H:%M:%S")
 
     return game_data
 
 
+def get_game_id_from_inputted_game(inputted_game: list,
+                                   cursor: connection.cursor) -> RealDictRow:
+    """Given a game that has already been inputted into the database, return its
+    matched game_id entry in the database."""
+    cursor.execute("""SELECT game_id FROM game
+                        WHERE name = %s""", (inputted_game[0],))
+
+    game_id = cursor.fetchone()['game_id']
+
+    return game_id
+
+
+def get_dev_id(input_game: list, cursor: connection.cursor) -> RealDictRow:
+    """Given a game, returns the corresponding developer_id entry if its developer 
+    name is in the developer table, and None otherwise."""
+
+    cursor.execute("""SELECT developer_id FROM developer
+                        WHERE developer_name = %s""", (input_game[3],))
+
+    developer_id_match = cursor.fetchone()
+
+    return developer_id_match
+
+
+def input_game_dev_get_dev_id(input_game: list, conn: connection) -> int:
+    """For each game in our input data, we check if its associated developer is in the
+    developer table. If not, we input it into the developer table.
+    Returns the developer_id entry from the given developer name."""
+
+    with conn.cursor() as cur:
+
+        developer_id_match = get_dev_id(input_game, cur)
+
+        if developer_id_match is None:
+
+            cur.execute("""INSERT INTO developer (developer_name)
+                    VALUES (%s)""", (input_game[3],))
+
+            developer_id_match = get_dev_id(input_game, cur)
+
+        cur.close()
```

Comment:
Don't believe this needs closing

Reply:
Apologies, I thought this was good practice! Will review this!